ext/scheduler/airflow/client.go

```diff
@@ -104,18 +104,20 @@ func buildEndPoint(host, path string) string {
 	return u.String()
 }
 
-func getJobRuns(res DagRunListResponse, spec *cron.ScheduleSpec) ([]*scheduler.JobRunStatus, error) {
+func getJobRuns(res DagRunListResponse, spec *cron.ScheduleSpec, withExternalTrigger bool) ([]*scheduler.JobRunStatus, error) {
 	var jobRunList []*scheduler.JobRunStatus
 	if res.TotalEntries > pageLimit {
 		return jobRunList, errors.InternalError(EntityAirflow, "total number of entries exceed page limit", nil)
 	}
 	for _, dag := range res.DagRuns {
-		if !dag.ExternalTrigger { // only include scheduled runs
-			scheduledAt := spec.Next(dag.ExecutionDate)
-			jobRunStatus, _ := scheduler.JobRunStatusFrom(scheduledAt, dag.State)
-			// use multi error to collect errors and proceed
-			jobRunList = append(jobRunList, &jobRunStatus)
+		if !withExternalTrigger && dag.ExternalTrigger {
```

Comment:
actually the if condition seems a little tricky to grasp , although it is correct , but maybe it can be simplified , by removing the continue statement and making it simple , we can discuss this

Reply:
yes agree, I just thought it's not understandable by anyone. instead of checking the condition for skip, might better to use the condition to check if it's allowed to proceed. much simple